src/routes/Signup.js

```diff
@@ -0,0 +1,169 @@
+import signupImage from "../images/signup-back.jpg";
+import Logo from "../images/logo.png";
+import "./SignupStyle.css";
+import { useState, useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+
+const Signup = () => {
+  const [username, setUsername] = useState("");
+  const [password, setPassword] = useState("");
+  const [confPassword, setConfPassword] = useState("");
+  const [users, setUsers] = useState([]);
+  const navigate = useNavigate();
+  const [checked, setChecked] = useState();
+
+  // Retrieve user data from API
+  useEffect(() => {
+    const fetchData = async () => {
+      const response = await fetch(
+        "https://655b7080ab37729791a91da3.mockapi.io/users/users"
+      );
+      const users = await response.json();
+      setUsers(users);
+      console.log(users);
+    };
+
+    fetchData();
+  }, []);
+
+  //
+
+  // Handle changes in inputs
+  const handleUsernameChange = (e) => {
+    setUsername(e.target.value);
+    // console.log(e.target.value);
+  };
+
+  const handlePasswordChange = (e) => {
+    setPassword(e.target.value);
+    // console.log(e.target.value);
+  };
+
+  const handleConfPasswordChange = (e) => {
+    setConfPassword(e.target.value);
+    // console.log(e.target.value);
+  };
+
+  // Sign up handler
+  const handleSignup = async (e) => {
+    e.preventDefault();
+
+    // Search for user with credentials from input fields
+    const loggedUser = users.find(
+      (user) => user.username === username && user.password === password
+    );
+
+    if (!loggedUser) {
+      // Make a post request to mock API
+      const response = await fetch(
+        "https://655b7080ab37729791a91da3.mockapi.io/users/users",
+        {
+          method: "POST",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({
+            username,
+            password,
+            role: "regular",
+            isPasswordSafe: true,
+          }),
+        }
+      );
```

Comment:
As a user if I add an already existing username but different password, the user is created anyway.

Ideal outcome:

The app checks only if the username exists. If it does the user should be notified in some way (e.g. Alert message)